homeassistant/components/gardena/vacuum.py

```diff
@@ -0,0 +1,174 @@
+    SUPPORT_STOP,
+    SUPPORT_START,
+    STATE_IDLE,
+    STATE_PAUSED,
+    STATE_CLEANING,
+    STATE_DOCKED,
+    STATE_RETURNING,
+    STATE_ERROR,
+    ATTR_BATTERY_LEVEL,
+)
+
+from . import GARDENA_LOCATION, GARDENA_CONFIG, CONF_MOWER_DURATION
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(minutes=1)
+
+SUPPORT_GARDENA = (
+    SUPPORT_BATTERY | SUPPORT_RETURN_HOME | SUPPORT_STOP | SUPPORT_START | SUPPORT_STATE
+)
+
+
+async def async_setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Neato vacuum."""
+    dev = []
+
+    for mower in hass.data[GARDENA_LOCATION].find_device_by_type("MOWER"):
+        dev.append(GardenaSmartMower(hass, mower, hass.data[GARDENA_CONFIG]))
+    _LOGGER.debug("Adding mower as vacuums %s", dev)
+    add_entities(dev, True)
+
+
+class GardenaSmartMower(StateVacuumDevice):
+    """Representation of a Gardena Connected Mower."""
+
+    def __init__(self, hass, mower, config):
+        """Initialize the Gardena Connected Mower."""
+        self._device = mower
+        self._config = config
+        self._name = "{}".format(self._device.name)
+        self._state = STATE_IDLE
+        self._error_message = ""
+
+    async def async_added_to_hass(self):
+        """Subscribe to events."""
+        self._device.add_callback(self.async_update_callback)
+
+    @property
+    def should_poll(self) -> bool:
+        """No polling needed for a vacuum."""
+        return False
+
+    @callback
+    def async_update_callback(self):
+        """Call update for home assistant when the device is updated."""
+        self.schedule_update_ha_state(True)
+
+    async def async_update(self):
+        """Update the states of Gardena devices."""
+        _LOGGER.debug("Running Gardena update")
+        # Managing state
+        state = self._device.state
+        _LOGGER.debug("Mower has state %s", state)
+        if state in ["WARNING", "ERROR", "UNAVAILABLE"]:
+            _LOGGER.debug("Mower has an error")
+            self._state = STATE_ERROR
+            self._error_message = self._device.last_error_code
+        else:
+            _LOGGER.debug("Getting mower state")
+            activity = self._device.activity
+            _LOGGER.debug("Mower has activity %s", activity)
+            if activity == "PAUSED":
+                self._state = STATE_PAUSED
+            elif activity in [
+                "OK_CUTTING",
+                "OK_CUTTING_TIMER_OVERRIDDEN",
+                "OK_LEAVING",
+            ]:
+                self._state = STATE_CLEANING
+            elif activity == "OK_SEARCHING":
+                self._state = STATE_RETURNING
+            elif activity in [
+                "OK_CHARGING",
+                "PARKED_TIMER",
+                "PARKED_PARK_SELECTED",
+                "PARKED_AUTOTIMER",
+            ]:
+                self._state = STATE_DOCKED
+            elif activity == "NONE":
+                _LOGGER.debug("Mower has no activity")
```

Comment:
No `self._state` here?

Reply:
I put `self._state = None` but I am not sure when activity can be equal to "NONE"